### ISSUE TITLE
vagrant rdp with windows 10 box does not work

### ISSUE THREAD

**Issue by mahmoud-ashi** [gusztavvargadr/packer#172]
When running `vagrant rdp` command, no thing happens at the moment.

**Comment by gusztavvargadr** [comment]
@mahmoud-ashi thanks for raising this, can you please let me know your host details (OS, version) and provider configuration (which virtualization provider, which version).

Can you please also try to include Vagrant's debug logs?

https://www.vagrantup.com/docs/other/debugging.html

**Comment by mahmoud-ashi** [comment]
Hey @gusztavvargadr. Here are the configurations:
Vagrant version: 2.2.7
Ruby version: 2.4.9
RubyGems version: 2.6.14.4
VirtualBox version: v6.1.4
Host OS: Ubuntu 18.04

[rdp.log](https://github.com/gusztavvargadr/packer/files/4340736/rdp.log)

Let me know if I can help.

**Comment by gusztavvargadr** [comment]
@mahmoud-ashi thanks, based on the error message in the logs `Failed to connect, CredSSP required by server` I've found this: https://www.syskit.com/blog/credssp-required-by-server-solutions/

I'd first try installing `xfreerdp` based on Vagrant's sources: https://github.com/hashicorp/vagrant/blob/85acf0cac724ef4bfda593a66e0c17f7e4110da0/plugins/hosts/linux/cap/rdp.rb#L8-L14

If not, another, but less safe option can be to turn of Network Level Authentication for RDP with Vagrant provisioning.

Please let me know if that helps.

**Comment by gusztavvargadr** [comment]
@mahmoud-ashi did you manage to solve it?

**Comment by mahmoud-ashi** [comment]
I did not resolve it. I have some other base box which works fine so I don't think it is related to my machine. I'm sorry I was not able to reply earlier.
I will play around again over the weekend and try to find the root cause.
Thanks for asking.

**Comment by gusztavvargadr** [comment]
@mahmoud-ashi there are newer versions now, including Version 2004 as well. Please let me know if you still encounter this or managed to resolve the issue.

https://app.vagrantup.com/gusztavvargadr/boxes/windows-10

**Comment by jvalrog** [comment]
Hello, I'm having the same problem.

I'm using Vagrant 2.2.9 on Ubuntu 20.04.
Xfreerdp: 2.1.1
Box: gusztavvargadr/windows-10 (virtualbox, 2004.0.2006)

**NOTE**: I can connect directly using xfreerdp as: `xfreerdp /u:vagrant /p:vagrant /v:localhost:33389`

Debug message:
```
...
...
[15:30:50:483] [83479:83480] [INFO][com.freerdp.core] - freerdp_tcp_connect:freerdp_set_last_error_ex resetting error state
Certificate details for 127.0.0.1:33389 (RDP-Server):
	Common Name: DESKTOP-SQRJAJ9
	Subject:     CN = DESKTOP-SQRJAJ9
	Issuer:      CN = DESKTOP-SQRJAJ9
	Thumbprint:  66:bb:0f:74:ff:ea:29:ea:21:eb:2e:70:6b:29:35:c9:4c:b1:36:37:2e:e9:19:e1:9f:07:ae:32:0c:50:6c:2c
The above X.509 certificate could not be verified, possibly because you do not have
the CA certificate in your certificate store, or the certificate has expired.
Please look at the OpenSSL documentation on how to add a private CA to the store.
Do you trust the above certificate? (Y/T/N) 
DEBUG subprocess: stderr: [15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
[15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - @           WARNING: CERTIFICATE NAME MISMATCH!           @
[15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
[15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - The hostname used for this connection (127.0.0.1:33389) 
[15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - does not match the name given in the certificate:
[15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - Common Name (CN):
[15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - 	DESKTOP-SQRJAJ9
[15:30:50:494] [83479:83480] [ERROR][com.freerdp.crypto] - A valid certificate for the wrong name should NOT be trusted!
```

The output is sometimes different:

```
DEBUG subprocess: stderr: [15:35:11:870] [83871:83872] [WARN][com.freerdp.crypto] - Certificate verification failure 'unable to get local issuer certificate (20)' at stack position 0
[15:35:11:870] [83871:83872] [WARN][com.freerdp.crypto] - CN = DESKTOP-SQRJAJ9
DEBUG subprocess: stderr: [15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
[15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - @           WARNING: CERTIFICATE NAME MISMATCH!           @
[15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
DEBUG subprocess: stdout: [15:35:11:537] [83871:83872] [INFO][com.freerdp.core] - freerdp_connect:freerdp_set_last_error_ex resetting error state
[15:35:11:537] [83871:83872] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[15:35:11:537] [83871:83872] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[15:35:11:537] [83871:83872] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[15:35:11:856] [83871:83872] [INFO][com.freerdp.primitives] - primitives autodetect, using optimized
[15:35:11:857] [83871:83872] [INFO][com.freerdp.core] - freerdp_tcp_is_hostname_resolvable:freerdp_set_last_error_ex resetting error state
[15:35:11:857] [83871:83872] [INFO][com.freerdp.core] - freerdp_tcp_connect:freerdp_set_last_error_ex resetting error state
Certificate details for 127.0.0.1:33389 (RDP-Server):
	Common Name: DESKTOP-SQRJAJ9
	Subject:     CN = DESKTOP-SQRJAJ9
	Issuer:      CN = DESKTOP-SQRJAJ9
	Thumbprint:  66:bb:0f:74:ff:ea:29:ea:21:eb:2e:70:6b:29:35:c9:4c:b1:36:37:2e:e9:19:e1:9f:07:ae:32:0c:50:6c:2c
The above X.509 certificate could not be verified, possibly because you do not have
the CA certificate in your certificate store, or the certificate has expired.
Please look at the OpenSSL documentation on how to add a private CA to the store.
Do you trust the above certificate? (Y/T/N) 
DEBUG subprocess: stderr: [15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - The hostname used for this connection (127.0.0.1:33389) 
[15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - does not match the name given in the certificate:
[15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - Common Name (CN):
[15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - 	DESKTOP-SQRJAJ9
[15:35:11:870] [83871:83872] [ERROR][com.freerdp.crypto] - A valid certificate for the wrong name should NOT be trusted!
```

**Comment by gusztavvargadr** [comment]
@jvalrog thanks for reporting this, though it does not seem to be exactly the same issue as above.

Can you try passing the parameter to ignore certificates to your client (`xfreerdp` I guess). Something like this could help? https://www.vagrantup.com/docs/cli/rdp.html

````
vagrant rdp -- /cert-ignore
```

I can try reproducing myself in the meantime.

**Comment by mahmoud-ashi** [comment]
@gusztavvargadr, sorry I have been very busy lately. I'll check it out over the weekend and report back.

**Comment by gusztavvargadr** [comment]
@mahmoud-ashi no worries, take your time of course :) My suggestion is still to try to install `xfreerdp`, based on your logs the `rdesktop` client is used by Vagrant as a fallback, which has known limitations.

If you manage to make it work with some other box, please share the details of your setup. Thanks!

**Comment by jvalrog** [comment]
@gusztavvargadr that command worked!

Is it normal for the VM to take several minutes to go up? Apparently it setups things at the start.

**Comment by gusztavvargadr** [comment]
@jvalrog great that RDP works now!

Yes, it is the Windows way of cleaning up an installation that I do during packaging, called [Sysprep](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/sysprep--generalize--a-windows-installation).

The default box though comes with very limited resources (only 1 GB RAM and 1 vCPU), you can try adjusting it in your Vagrant configuration, for Windows 10 at least 4 GB RAM and 2 vCPU that I recommend. This will impact the boot time as well.

Depending on your scenario you can try the LTSC version that is slimmer and boots faster, or Windows Server.

https://app.vagrantup.com/gusztavvargadr/boxes/windows-10-enterprise-ltsc
https://app.vagrantup.com/gusztavvargadr/boxes/windows-server

**Comment by jvalrog** [comment]
@gusztavvargadr great to know, I'll try it next. Thanks for an amazing work.

**Comment by mahmoud-ashi** [comment]
I tried the suggestion from @gusztavvargadr and it worked. First I uninstalled `rdesktop` and installed `xfreerdp`. Next, when I ran the command `vagrant rdp`, I faced the same issue as @jvalrog [here](https://github.com/gusztavvargadr/packer/issues/172#issuecomment-649590602) (certificate not verified). When I ran the command `vagrant rdp -- /cert-ignore` as @gusztavvargadr mentioned earlier, it worked fine.
